tt_metal/programming_examples/single_thread_eltwise_binary/single_thread_eltwise_binary.cpp

```diff
@@ -0,0 +1,205 @@
+        std::shared_ptr<tt::tt_metal::Buffer> src1_dram_buffer = CreateBuffer(dram_config);
+        std::shared_ptr<tt::tt_metal::Buffer> dst_dram_buffer = CreateBuffer(dram_config);
+        // Since all interleaved buffers have size == page_size, they are entirely contained in the first DRAM bank
+        uint32_t src0_bank_id = 0;
+        uint32_t src1_bank_id = 0;
+        uint32_t dst_bank_id = 0;
+        /*
+         * Use circular buffers to set input and output buffers that the
+         * compute engine will use.
+         */
+        constexpr uint32_t src0_cb_index = tt::CBIndex::c_0;
+        constexpr uint32_t num_input_tiles = block_size * 2;
+        CircularBufferConfig cb_src0_config =
+            CircularBufferConfig(num_input_tiles * single_tile_size, {{src0_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(src0_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_src0_config);
+
+        constexpr uint32_t src1_cb_index = tt::CBIndex::c_1;
+        CircularBufferConfig cb_src1_config =
+            CircularBufferConfig(num_input_tiles * single_tile_size, {{src1_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(src1_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_src1_config);
+
+        constexpr uint32_t output_cb_index = tt::CBIndex::c_16;
+        constexpr uint32_t num_output_tiles = block_size * 2;
+        CircularBufferConfig cb_output_config =
+            CircularBufferConfig(num_output_tiles * single_tile_size, {{output_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(output_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_output_config);
+
+        /*
+         * Specify data movement kernels for reading/writing data to/from
+         * DRAM.
+         */
+        KernelHandle binary_reader_kernel_id = CreateKernel(
+            program,
+            "tt_metal/kernels/dataflow/reader_dummy.cpp",
+            core,
+            DataMovementConfig{.processor = DataMovementProcessor::RISCV_1, .noc = NOC::RISCV_1_default});
+
+        KernelHandle unary_writer_kernel_id = CreateKernel(
+            program,
+            "tt_metal/kernels/dataflow/writer_dummy.cpp",
+            core,
+            DataMovementConfig{.processor = DataMovementProcessor::RISCV_0, .noc = NOC::RISCV_0_default});
+
+        /*
+         * Set the parameters that the compute kernel will use.
+         */
+        std::vector<uint32_t> compute_kernel_args = {};
+
+        constexpr bool fp32_dest_acc_en = false;
+        constexpr bool math_approx_mode = false;
+
+        /*
+         * Use the add_tiles operation available in the eltwise_binary
+         * compute kernel.
+         */
+        KernelHandle eltwise_binary_kernel_id = CreateKernel(
+            program,
+            "tt_metal/kernels/compute/single_thread_eltwise_binary_with_read_write.cpp",
```

Comment:
For kernels within examples, please put them under the example's folder and heavily annotate them. This way newcomers won't have to dig through the entire project to find information.